galsim/chromatic.py

```diff
@@ -1359,3 +1770,76 @@ def withScaledFlux(self, flux_ratio):
             return ChromaticAutoCorrelation( self.obj * math.sqrt(flux_ratio), **self.kwargs )
         else:
             return ChromaticObject(self).withScaledFlux(flux_ratio)
+
+class ChromaticOpticalPSF(ChromaticObject):
+    """A subclass of ChromaticObject meant to represent chromatic optical PSFs.
+
+    Chromaticity plays two roles in optical PSFs. First, it determines the diffraction limit, via
+    the wavelength/diameter factor.  Second, aberrations such as defocus, coma, etc. are typically
+    defined in physical distances, but their impact on the PSF depends on their size in units of
+    wavelength.  Other aspects of the optical PSF do not require explicit specification of their
+    chromaticity, e.g., once the obscuration and struts are specified in units of the aperture
+    diameter, their chromatic dependence gets taken care of automatically.
+
+    When using interpolation to speed up image rendering (see ChromaticObject.setupInterpolation()
+    method for details), including ~10-15 samples across any given bandpass should be sufficient.
+    For moderate accuracy, 5 is often sufficient.  Likewise, for high accuracy, `oversample_fac`
+    should be in the range 1.5-2, whereas for moderate accuracy, 1 is often sufficient.  All of
+    these statements assume that aberrations are not very large (typically <~0.25 waves, which is
+    commonly satisfied by space telescopes); if they are larger than that, then more stringent
+    settings are required.
+
+    Also, note that in order to render an image of a real physical object, the ChromaticOpticalPSF
+    should be convolved with some object that has an SED.  Drawing without doing so corresponds to
+    using a flat SED.
```

Comment:
OK, I think I might see your point here.  You are pointing out that this is not the correct SB profile of a _star_, since star's have an SED that is not included here.  

I think the point you are making is already obvious with respect to galaxies -- that if you use a regular GSObject, then it corresponds to a flat SED.  But maybe it's not as obvious that stars also have SEDs that are not included here in the PSF object.  So here is a proposed paragraph that maybe makes this point more clearly:

Note that a ChromaticOpticalPSF by itself is NOT the correct thing to use to draw an image of a
star.  Stars (and galaxies too, of course) have an SED that is not flat.  To draw a real star, you
should either multiply the ChromaticOpticalPSF object by an SED, or convolve it by a point
source (typically approximated by a very tiny Gaussian) multiplied by an SED:

    >>> psf = galsim.ChromaticOpticalPSF(...)
    >>> star = galsim.Gaussian(sigma = 1.e-6) * psf_sed
    >>> final_star = galsim.Convolve( [psf, star] )
    >>> final_star.drawImage(bandpass = bp, ...)